eventually/static/src/containers/topicItemList/EditItemDialog.js

```diff
@@ -0,0 +1,147 @@
+
+    handleDescription = event => {
+        const regex = /^.{10,1024}$/;
+        if(regex.test(event.target.value) === true ) {
+            this.setState({
+                messageDescription: '',
+                description: event.target.value,
+                descriptionIsValid: true,
+            });
+        } else {
+            this.setState({
+                messageDescription: 'Required a minimum length of 10 characters, maximum length of 1024 characters',
+                descriptionIsValid: false,
+            });
+        }
+    };
+
+    handleChange = (event, index, value) => this.setState({'form': value});
+
+    handleSubmit = () => {
+        if(this.state.nameIsValid & this.state.descriptionIsValid){
+            const data = {
+                'name': this.state.name,
+                'description': this.state.description,
+                'form': this.state.form
+            };
+            putEditItemService(
+                this.props.curriculumId, this.props.topicId, this.props.id, data
+            ).then(response => {
+                this.props.getItemList();
+                this.handleClose();
+            });
+
+        }
+    };
+
+    render() {
+        const disable = !(this.state.nameIsValid && this.state.descriptionIsValid);
+        const actions = [
+            <FlatButton
+                label="Cancel"
+                primary={true}
+                onClick={this.handleClose}
+            />,
+            <FlatButton
+                label="Submit"
+                primary={true}
+                onClick={this.handleSubmit}
+                disabled={disable}
+            />,
+        ];
+        return (
+            <div>
+                <RaisedButton
+                    icon={<ModeEdit />}
+                    onClick={this.handleOpen} />
+                <Dialog
+                    title="Edit item"
+                    actions={actions}
+                    modal={false}
+                    open={this.state.open}
+                    onRequestClose={this.handleClose}
+                >
+                    <TextField
+                        floatingLabelText="Name"
+                        defaultValue={this.state.name}
+                        fullWidth={true}
+                        onChange={this.handleName}
+                        errorText={this.state.messageName} />
+                    <TextField
+                        floatingLabelText="Description"
+                        defaultValue={this.state.description}
+                        onChange={this.handleDescription}
+                        errorText={this.state.messageDescription}
+                        multiLine={true}
+                        rowsMax={4}
+                        fullWidth={true} />
+                    <SelectField
+                        floatingLabelText="Select form"
+                        value={this.state.form}
+                        onChange={this.handleChange}
+                    >
```

Comment:
` onChange={this.handleChange}
                    >`
to
` onChange={this.handleChange}>`

Reply:
done